File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        l = cur_cost = res = 0
        for r in range(0, len(s)):
            cur_cost += abs(ord(s[r]) - ord(t[r]))
            while l <= r and cur_cost > maxCost:
                cur_cost -= abs(ord(s[l]) - ord(t[l]))
                l += 1
            res = max(res, r - l + 1)
        return res

File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.js
/**
 * @param {string} s
 * @param {string} t
 * @param {number} maxCost
 * @return {number}
 */
var equalSubstring = function(s, t, maxCost) {
    const n = s.length
    let i = 0, curCost = 0, len = 0
    for (let j=0; j < n; j++) {
        curCost += Math.abs(s[j].charCodeAt(0) - t[j].charCodeAt(0))
        while (i <= j && curCost > maxCost) {
            curCost -= Math.abs(s[i].charCodeAt(0) - t[i].charCodeAt(0))
            i++
        }
        len = Math.max(len, j - i + 1)
    }
    return len
};